compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -803,6 +809,20 @@ impl<'a> Parser<'a> {
             self.expect_keyword(kw::Const)?;
             let span = tilde.to(self.prev_token.span);
             self.sess.gated_spans.gate(sym::const_trait_impl, span);
+            Some(span)
+        } else if self.eat_keyword(kw::Const) {
+            let span = self.prev_token.span;
+            self.sess.gated_spans.gate(sym::const_trait_impl, span);
+
+            self.struct_span_err(span, "const bounds must start with `~`")
```

Comment:
I'm not sure if `~const Trait` is called  a "const" bound technically-- I would call it a "const if const" bound. But whatever, should be fine.